src/ncp/ncp_spinel.cpp

```diff
@@ -102,6 +112,106 @@ otbrError NcpSpinel::SpinelDataUnpack(const uint8_t *aDataIn, spinel_size_t aDat
     return error;
 }
 
+void NcpSpinel::DatasetSetActiveTlvs(const otOperationalDatasetTlvs &aActiveOpDatasetTlvs, AsyncTaskPtr aAsyncTask)
+{
+    otError      error        = OT_ERROR_NONE;
+    EncodingFunc encodingFunc = [this, &aActiveOpDatasetTlvs] {
+        return EncodeDatasetSetActiveTlvs(aActiveOpDatasetTlvs);
+    };
+
+    VerifyOrExit(mDatasetSetActiveTask == nullptr);
+    SuccessOrExit(error = SetProperty(SPINEL_PROP_THREAD_ACTIVE_DATASET, encodingFunc));
+    mDatasetSetActiveTask = aAsyncTask;
+
+exit:
+    if (error != OT_ERROR_NONE)
+    {
+        mTaskRunner.Post([aAsyncTask, error](void) { aAsyncTask->SetResult(error, "Failed to set active dataset!"); });
+    }
+}
+
+void NcpSpinel::Ip6SetEnabled(bool aEnable, AsyncTaskPtr aAsyncTask)
+{
+    otError      error        = OT_ERROR_NONE;
+    EncodingFunc encodingFunc = [this, aEnable] { return mEncoder.WriteBool(aEnable); };
+
+    VerifyOrExit(mIp6SetEnabledTask == nullptr);
+
+    error              = SetProperty(SPINEL_PROP_NET_IF_UP, encodingFunc);
+    mIp6SetEnabledTask = aAsyncTask;
+
+exit:
+    if (error != OT_ERROR_NONE)
+    {
+        mTaskRunner.Post(
+            [aAsyncTask, error](void) { aAsyncTask->SetResult(error, "Failed to enable the network interface!"); });
+    }
+    return;
+}
+
+void NcpSpinel::ThreadSetEnabled(bool aEnable, AsyncTaskPtr aAsyncTask)
+{
+    otError      error        = OT_ERROR_NONE;
+    EncodingFunc encodingFunc = [this, aEnable] { return mEncoder.WriteBool(aEnable); };
+
+    VerifyOrExit(mThreadSetEnabledTask == nullptr);
+
+    error                 = SetProperty(SPINEL_PROP_NET_STACK_UP, encodingFunc);
+    mThreadSetEnabledTask = aAsyncTask;
+
+exit:
+    if (error != OT_ERROR_NONE)
+    {
+        mTaskRunner.Post(
+            [aAsyncTask, error](void) { aAsyncTask->SetResult(error, "Failed to enable the Thread network!"); });
+    }
+    return;
+}
+
+void NcpSpinel::ThreadDetachGracefully(AsyncTaskPtr aAsyncTask)
+{
+    otError      error        = OT_ERROR_NONE;
+    EncodingFunc encodingFunc = [] { return OT_ERROR_NONE; };
+
+    VerifyOrExit(mThreadDetachGracefullyTask == nullptr);
```

Comment:
When this check fails, it returns an `OT_ERROR_NONE` without setting the result. Is this expected?

Reply:
Set to `OT_ERROR_BUSY` now.